ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -1,11 +1,134 @@
+  > Latest Snapshot:                      *
+  > LedgerDB:                                 C |> D > E > F > ... > T
+
+  If the latest snapshot that was taken corresponds to B, if the node is
+  shutdown in this instant, when it restarts, it will load the following state
+  from disk:
+
+  >                                              |<---  k blocks --->|
+  > Selected Chain: Genesis |> ... > A > B >  C  > D > E > F > ... > T
+  > ImmutableDB:    Genesis |> ... > A > B >  C
+  > VolatileDB:                               C |> D > E > F > ... > T  (# blocks [D,T] <= k)
+  > LedgerDB:                            B |>
+
+  Proceeding then to replay the blocks on the ImmutableDB and then the ones on
+  the VolatileDB (the latter will be validated when applied), reaching in the
+  end the same state as above.
+
+  Because each LedgerState is fully contained, there's no problem on the above
+  logic, even if we have some spans of time in which a shutdown will make the
+  node start from an older point.
+
+  It is responsibility of the
+  @Ouroboros.Conensus.Storage.ChainDB.Impl.Background.copyAndSnapshotRunner@
+  copying the blocks to the disk and when dictated by the @DiskPolicy@, take a
+  snapshot. Said snapshot will /always/ be taken from the current anchor of the
+  anchored sequence, as said block is put in the ImmutableDB by the runner
+  before taking the snapshot.
+
+  The @DiskPolicy@ will specify how many snapshots to keep around, as it is
+  possible to start from an older snapshot (due to them being self contained),
+  and also how much time we want at least between snapshots, in order to avoid
+  flushing too quickly.
+
+  New-style Snapshot and flushing
+  ===============================
+
+  In the new-style database we hace mainly 3 moving pieces: the sequence of
+  @ExtLedgerState blk EmptyMK@ states, the sequence of @TableDiff@s and the disk
+  backend.
+
+  Ledger states no longer contain a full UTxO and therefore they are dependent
+  on the history before them to make a consistent view of the UTxO after
+  applying a block. The list of can no longer be pruned to @k@ as we cannot lose
+  the sequence of differences between what was last flushed and the @length -
+  k@th block.
+
+  So adapting the example above, we can have the following situation:
+
+  >                                                |<---  k blocks --->|
+  > Selected Chain:   Genesis |> ... > A >  B > C  > D > E > F > ... > T
+  > ImmutableDB:      Genesis |> ... > A >  B > C
+  > VolatileDB:                                 C |> D > E > F > ... > T  (# blocks [D,T] <= k)
+  > DiskBackend:                       *
+  > LedgerStates:                      A |> B > C >  D > E > F > ... > T
+  > LedgerDiffs:                            B > C >  D > E > F > ... > T
+
+  Note that the LedgerDiffs is a list and not an anchored sequence, and
+  therefore can be empty.
+
+  * Flushing :: applying a sequence of differences to the disk backend to make
```

Comment:
These definitions seem a bit disconnected from the text. Maybe we want to place them before the example.